internal/colerrs/README.md

```diff
@@ -0,0 +1,17 @@
+# Internal collector errors
+
+This package includes helpers propagating errors from exporters or processors
+back to receivers. Error propagation is helpful in a setup where the Collector
+acts as a simple gateway between clients and other Collector instances or
+backends.
+
+Error propagation only works when there's synchronous processing of a pipeline.
```

Comment:
I guess I would be interested to know what would be the expected behaviour in these cases?

Does the propagation of errors stop once it reaches this types of components?
In that case, should a future improvement be to include a means of async notify?
Not sure of the benefit beyond, "Hey, maybe you should slow down" but I am sure if we make it available it would be handy for those components.

Reply:
Not sure if this would be possible, since the client would have already completed its transaction. If a client sends a bunch of telemetry that's batched, and the batch is then rejected, where would the error response go?